pkg/trace/agent/agent.go

```diff
@@ -159,6 +160,11 @@ func (a *Agent) loop() {
 // Process is the default work unit that receives a trace, transforms it and
 // passes it downstream.
 func (a *Agent) Process(t pb.Trace) {
+	now := time.Now()
+	defer func() {
```

Comment:
Note that the body of this function is only evaluated when the `defer` runs, after the function returns. This means that the variables captured in the closure (such as `now`) will have the same value that they do when the function returns. In other words, `now` will have the value from its second assignment further down on line 245. One way to avoid this is:

```go
defer func(start time.Time) {
		metrics.Timing("datadog.trace_agent.receiver.process_single_trace_ms", time.Since(start), nil, 1)
}(time.Now())
```
The function arguments are evaluated at the time of definition and will have the desired outcome.